docs/contributing-to-registry.md

```diff
@@ -0,0 +1,66 @@
+
+### Creating a directory for the new module
+<!-- TODO: need to discuss the pattern of the module path -->
+Add a new directory under the `modules` folder in your local azure-quickstart-templates repository with the path following the pattern `<ResourceProviderName>/<ModuleName>/<ModuleMajorVersion>.<ModuleMinorVersion>`. For examples: 
+- `Microsoft.Storage/storageAccounts/0.9`
+- `Microsoft.Web/webApplications/1.0`
+
+### Generating module files
+Open a terminal and navigate to the newly created folder. From there, run the following command to generate the required files for the Bicep public registry module:
+```
+bicep-registry-module generate
+```
+
+You should be able to see these files created in the module folder:
+- `metadata.json`
+  - The module manifest file that contains metadata you must provide, such as `description`.
+- `main.bicep`
+  - An empty Bicep file to be updated.
+- `azuredeploy.json`
+  - The ARM template file generated from `main.bicep`. This is the file that will be published to the Bicep public registry. You should not modify this file.
+- `azuredeploy.parameters.json`
+  - The ARM template parameters file for `azuredeploy.json`. The file is generated for running template deployment validation. You must provide values for the required parameters in the file.
+- `README.md`
+  - The README file generated based on the contents of `metadata.json` and `main.bicep`. Do not update this file manually.
+- `version.json`
+  - The version file, together with the `azuredeploy.json` file, are used to calculate the patch version of the module automatically when publishing the module to the Bicep public module registry. You should not edit this contents of the file.
+
+### Authoring module files
+The only two files that you need to edit are `metadata.json` and  `main.bicep`. When authoring `main.bicep`, make sure to provide a description for each parameter and output. You are free to create other Bicep files and reference them as local modules in `main.bicep` if needed. Once you are done, run `bicep-registry-module generate` again to refresh the other files. You may need to update `azuredeploy.parameters.json` to provide parameter values after the generation if there are changes to the template parameters.
+
+## Updating an existing module
+To update an existing module, simply follow the [Authoring module files](#authoring-module-files) section to update and regenerate the module files.
+
+## Validating a module
+
+> Before running the command, it is recommended to run `generate` and check `azuredeploy.parameters.json` to ensure all files are up-to-date.
+
+You may use the bicep-registry-module .NET tool to validate files in a registry module. To do so, invoke the follow command from the module folder:
+```
+bicep-registry-module validate
```

Comment:
The tool name is really long. Should we shorten it to make it quicker to type? Maybe `biceprm`, `bicepreg`, `brm` or some variation of that? (Can be done in a different PR.)

Reply:
Good point. The fun fact is that I set the name to `brm` temporarily when I was testing my code. When I submitted the PR I changed it to the long name. I think I like `brm` since it requires less typing. I'll send a follow up PR to change it.